### PR TITLE
Add multiple placement options for multinode uniconfig deployment

### PR DESCRIPTION
**Pull request by Jozefiel**
- change default single uc deployment from node.id to node.role
 - add role, label, hostname options to generte_uc_compose.sh
 - use x-common-settings in uc composefile

### CODE REVIEW

File: generate_uc_compose.sh [hunk 1]
```diff
@@ -98,6 +141,18 @@ do
...
     shift
 done
+
+if [[ -z $TYPE ]]; then
+  echo -e "${ERROR} Node placement type not defined. See help !!!"
+  exit 1
+fi
+
+if [[ -z $__SERVICE_NAME_DEFINED ]]; then
+  echo -e "${ERROR}Service name not defined. See help !!!"
```
Comment (marosmars) [inline generate_uc_compose.sh:151]: space missing

File: composefiles/swarm-uniconfig.yml [hunk 2]
```diff
@@ -18,6 +18,11 @@ x-logging: &logging_loki
         - action: labelmap
           regex: swarm_(service)
 
+x-placement: &placement
+      placement:
+        constraints:
+          - node.role == manager
```
Comment (marosmars) [inline composefiles/swarm-uniconfig.yml:24]: should uniconfig be on manager node ?

File: generate_uc_compose.sh [hunk 3]
```diff
@@ -115,10 +170,56 @@ function checkUniconfigServiceName {
...
+  if [[ "${node}" == "${1}" ]]; then
+      echo -e "${INFO} Node ${message} ${bold}${1}${normal} is in swarm - Hostname: ${bold}${node_hostname}${normal}"
   else 
-      echo -e "${WARNING} Node ${1} not in swarm"
+      echo -e "${SWARM_STATUS} Node ${message} ${bold}${1}${normal} not in swarm"
+      if [[ $__FORCE_GENERATE == 'false' ]]; then
+        echo -e "${SWARM_STATUS} Composefiles was not generated"
+        exit 1
+      fi
+      echo -e "${SWARM_STATUS} Composefiles was force-generated with wrong placement settings"
```
Comment (marosmars) [inline generate_uc_compose.sh:222]: was -> were

File: composefiles/swarm-uniconfig.yml (hunk 2 above)
Comment (Jozefiel) [reply composefiles/swarm-uniconfig.yml:24]: Default settings are for single-node deployment and all services are running on a master node.
This parameter is replaced during generation compose files by selected placement type.

**Pull request merged**